File: demo/src/main/java/at/htlsaalfelden/main/Horserace/Horse.java
package at.htlsaalfelden.main.Horserace;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Random;

public class Horse {
    private String names[] = {"Thunderhoof", "Shadowrunner", "Starlight Gallop", "Midnight Mirage", "Silver Mane"};
    private String name;
    private float speed;
    private HashMap<String, Float> bets = new HashMap<>();
    private float runDistance = 0;
    private Random random = new Random();

    public Horse(){
        this.name = names[random.nextInt(5)];
        this.speed = 50 + random.nextFloat(10);
    }

    @Override
    public String toString(){
        return "Name: " + this.name + " speed: "+ this.speed + " bets: " + this.bets.size();
    }

    public String getName(){
        return this.name;
    }

    public void setName(String newName){
        this.name = newName;
    }

    public float getSpeed(){
        return this.speed;
    }

    public void addBet(float f, String better){
        this.bets.put(better, f);
    }

    public HashMap<String, Float> getBets(){
        return this.bets;
    }

    public float getRunDistance(){
        return  this.runDistance;
    }


    public void addRunDistance(float f){
        this.runDistance += f;
    }
}

File: webclient_maier/src/Utitlity/Horse.ts
export default class Horse {
  private _name: string;
  private _speed: number;
  private _bets: Record<string, number>; // Assuming bets is a mapping of string to number
  private _runDistance: number = 0;

  constructor(name: string, speed: number, bets: Record<string, number> = {}) {
    this._name = name;
    this._speed = speed;
    this._bets = bets;
  }

  public get name(){
    return this._name;
  }

  public set name(name:string){
    this._name = name;
  }

  public get speed(){
    return this._speed;
  }

  public set speed(speed:number){
    this._speed = speed;
  }

  public get bets(): Record<string, number> {
    return this._bets;
  }

  public set bets(bets: Record<string, number>){
    this.bets = bets;
  }

  public get runDistance(){
    return this._runDistance;
  }

  public set runDistance(distance:number){
    this._runDistance = distance;
  }

}
